webapp/packages/plugin-user-profile/src/UserProfileForm/UserAuthenticationPart/ChangePassword.tsx

```diff
@@ -5,20 +5,56 @@
  * Licensed under the Apache License, Version 2.0.
  * you may not use this file except in compliance with the License.
  */
+import { observable } from 'mobx';
 import { observer } from 'mobx-react-lite';
 
-import { Form, Group, GroupTitle, InputField, useCustomInputValidation, usePasswordValidation, useTranslate } from '@cloudbeaver/core-blocks';
-import { isValuesEqual } from '@cloudbeaver/core-utils';
+import { UserInfoResource } from '@cloudbeaver/core-authentication';
+import {
+  ColoredContainer,
+  ConfirmationDialog,
+  Container,
+  Form,
+  type FormChangeValues,
+  Group,
+  GroupTitle,
+  InputField,
+  ToolsAction,
+  ToolsPanel,
+  useCustomInputValidation,
+  useExecutor,
+  useForm,
+  useObservableRef,
+  usePasswordValidation,
+  useTranslate,
+} from '@cloudbeaver/core-blocks';
+import { useService } from '@cloudbeaver/core-di';
+import { CommonDialogService, DialogueStateResult } from '@cloudbeaver/core-dialogs';
+import { NotificationService } from '@cloudbeaver/core-events';
+import { ExecutorInterrupter } from '@cloudbeaver/core-executor';
+import { isObjectsEqual, isValuesEqual } from '@cloudbeaver/core-utils';
 
+import { userProfileContext } from '../../userProfileContext.js';
+import { UserProfileOptionsPanelService } from '../../UserProfileOptionsPanelService.js';
 import type { IUserProfileFormAuthenticationState } from './IUserProfileFormAuthenticationState.js';
 
-interface Props {
-  state: IUserProfileFormAuthenticationState;
-  disabled?: boolean;
-}
+const INITIAL_STATE: IUserProfileFormAuthenticationState = {
+  oldPassword: '',
+  password: '',
+  repeatedPassword: '',
+};
```

Comment:
Please inline it in `useObservableRef` and also check other usages as references (these fields will be mutated)